### PR TITLE
Anime - Revived PR

### PR DESCRIPTION
**Pull request by iambibhas**
This is a revived version of #1664. Fixed the complaints on the other PR. Removed all the custom CSS and using the `movies` template.

selection_042

------

IA Page: https://duck.co/ia/view/anime

### CODE REVIEW

File: share/spice/anime/anime.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    cover_image = DDG.toHTTP(item.cover_image);
+                }
+
+                return {
+                    img: cover_image,
+                    image: cover_image,
+                    img_m: cover_image,
+                    heading: item.title,
+                    title: item.title,
+                    abstract: item.synopsis.split(' ').slice(0,49).join(' ') + '...',
```
Comment (moollaza) [inline share/spice/anime/anime.js]: The template handles shortening the content. We can just pass along the full synopsis and let the template deal with it.

File: share/spice/anime/anime.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    rating: item.community_rating,
+                    url: item.url,
+                    is_retina: (DDG.is3x || DDG.is2x) ? "is_retina" : "no_retina"
```
Comment (moollaza) [inline share/spice/anime/anime.js]: doesn't look like this get's used anywhere?

**Comment by iambibhas** [comment]
@moollaza Fixed those 2. But there seems to be a design issue on mobile devices 
selection_046

**Comment by moollaza** [comment]
@iambibhas I would just ditch the view on MyAnimeList button. There should really only be 1 button.

You could try changing the wording to "View on Hummingbird" and "View on MyAnimeList"

**Comment by iambibhas** [comment]
@moollaza MyAnimeList is kind of the go to place for anime lovers for references of an anime, hence wanted to keep it. Lemme try removing that word..

**Comment by abeyang** [comment]
I agree with @moollaza -- can we just reduce this down to one button?

**Comment by iambibhas** [comment]
@moollaza @abeyang Done. I forgot that the hummingbird url was already there as anchor on the title. Now there is just one button.

File: lib/DDG/Spice/Anime.pm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Anime.pm";
+topics "entertainment", "everyday";
+category "entertainment";
+attribution github => ['https://github.com/iambibhas','Bibhas'],
+           twitter => ['https://twitter.com/bibhasdn','Bibhas D'];
+
+spice proxy_cache_valid => "200 30d";
+spice to => 'https://hummingbird.me/api/v1/search/anime?query=$1';
+
+triggers startend => 'anime', 'hummingbird';
```
Comment (moollaza) [inline lib/DDG/Spice/Anime.pm]: I'm a little concerned with the `anime` trigger, given how broad that term is. A quick look at the autocomplete text when you type `anime` shows a lot of queries that trigger this which aren't searches for specifc anime books. Also, most of them return really irrelevant results so I think we'll need to use a relevancy block.

`anime wallpapers` for example returns tons of bad results..

Comment (iambibhas) [inline lib/DDG/Spice/Anime.pm]: Relevancy block on the front end might not be a good idea as there are literally thousands of animes and each of them have both japanese and english names, where the relevancy check will actually fail. Maybe we could have a list of stop words(taken from autocomplete) in the backend?

Comment (moollaza) [inline lib/DDG/Spice/Anime.pm]: @iambibhas sorry just saw this! I would suggest we use both methods to ensure we don't make redundant API calls, and only show relevant results to users

Comment (iambibhas) [inline lib/DDG/Spice/Anime.pm]: Added some stop words and relevancy block.

**Comment by jagtalon** [comment]
@iambibhas cool, checking it again!

**Comment by jagtalon** [comment]
@iambibhas updated the templates to hide the "0 reviews" that we're seeing. Please add `hideReviewText: true` to the `options` object. :)

**Comment by jagtalon** [comment]
Up on https://comserv.duckduckgo.com/?q=naruto+anime&ia=anime! @abeyang is currently reviewing it.  Also please change the API endpoint from HTTPS to HTTP (a little faster that way, I think)

**Comment by iambibhas** [comment]
@jagtalon -

> Please add `hideReviewText: true` to the `options` object. :)

Only if there is no review?

> Also please change the API endpoint from HTTPS to HTTP

Shouldn't we prefer HTTPS for almost everything? Just curious.

**Comment by jagtalon** [comment]
@iambibhas when it comes to endpoints it doesn't really matter because it's all proxied by us so if it's HTTP it will be always HTTPS for the users because it goes through us first. It's also a weee bit faster on our end. :)

**Comment by jagtalon** [comment]
@iambibhas there are no reviews for everything right?

**Comment by iambibhas** [comment]
@jagtalon done.

**Comment by jagtalon** [comment]
Hey @iambibhas. @abeyang suggested that we get rid of the detail pane since the data is a bit underwhelming. So now clicking on the tile should take you to the site itself.

What do you think?

**Comment by iambibhas** [comment]
@jagtalon @abeyang it has episode count, each episode length, rating, gist and links to MyAnimeList. It's "good to have" in my opinion.

**Comment by abeyang** [comment]
@iambibhas @jagtalon it's not so much that it's underwhelming -- I do think it's good to have -- it's that it would take a lot of work to just get it to look right. Internally, we're working on the detail pane templates ("DPT" -- which is coming along quite nicely!), so my suggestion is to hold off on this for now, and revisit once the DPT is finished.

**Comment by iambibhas** [comment]
@abeyang it looks good now. How about we let it roll now and revisit when new DPT is deployed and it starts looking ugly? We'll anyway have to revisit a lot more spices when the new DPT is here, right? Is the DPT coming very soon? Then it makes sense.

Just curious, what makes it hard for this one to look good on the new template? I'm sure there are other IA with details panes with even lesser amount of data.

**Comment by abeyang** [comment]
@iambibhas I'm fine with releasing this as-is, and revisiting once the new DPT is released. /cc @moollaza 

The whole point is that it should be **easy** for this (as well as all detail panes) to look good when the new template comes out, not harder! We've had a tough time in the past getting some DP's to look good, which is why we're doing the whole DPT thing.

**Comment by moollaza** [comment]
@abeyang alright sounds good to me! :+1:

**Comment by moollaza** [comment]
@iambibhas congrats! It's live: https://duckduckgo.com/?q=naruto+anime&ia=anime

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**